### PR TITLE
filterx: add `format_json()`

### PR DESCRIPTION
**Pull request by alltilla**

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Review by MrAnno — APPROVED** [review]

**Comment by MrAnno** [comment]
Note that "this kind of `format-json()`" does not produce an object or array when passing scalar parameters.
This is valid according to RFC 7159/RFC8259.

File: lib/filterx/object-json-array.c [hunk 1]
```diff
@@ -49,7 +49,6 @@ _truthy(FilterXObject *s)
 static gboolean
 _marshal_to_json_literal(FilterXJsonArray *self, GString *repr, LogMessageValueType *t)
 {
-  g_string_truncate(repr, 0);
```
[the remark this answers is not in the document]
Comment (alltilla) [reply lib/filterx/object-json-array.c:52]: I guess we should not have `marshal_append()` at all as we now have a `repr()` function. This PR was crafted before we had that idea. I think I will remove `marshal_append()` and replace its call with a `repr_append()`. What do you think?

[the remark this answers is not in the document]
Comment (MrAnno) [reply lib/filterx/object-json-array.c:52]: Okay, sounds good. `marshal_append()` looked good to me as well, I was just referring to this local `_marshal_to_json_literal()` function.

[the remark this answers is not in the document]
Comment (alltilla) [reply lib/filterx/object-json-array.c:52]: https://github.com/syslog-ng/syslog-ng/pull/4891

**Pull request merged**